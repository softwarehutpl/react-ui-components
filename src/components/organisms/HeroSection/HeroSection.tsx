import React from 'react';
import styled from 'styled-components';
import get from 'lodash/get';

type Capture = {
  title?: string;
  width?: number;
  description?: string;
  backgroundColor?: string;
  titleColor?: string;
  descriptionColor?: string;
  titleSize?: number;
  centerCapture?: boolean;
}

interface HeroSectionProps {
  width: string;
  height: string;
  backgroundColor?: string;
  backgroundImage?: string;
  horizontalPadding?: number;
  capture?: Capture;
}

interface HeroSectionWrapperProps {
  backgroundColor?: string;
  backgroundImage?: string;
  showTransitionEffect?: boolean;
  transitionEffect?: string;
  opacity?: number;
  width: string;
  height: string;
  horizontalPadding?: number;
}

export const HeroSectionWrapper = styled.div<HeroSectionWrapperProps>`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: ${({ width }) => width};
  height: ${({ height }) => height};
  padding: ${({ horizontalPadding }) => `0 ${horizontalPadding}px`};
  position: relative;
`;

export const HeroSectionImage = styled.img`
  width: 100%;
  height: 100%;
  object-fit: contain;
`;

export const HeroSectionCapture = styled.div<any>`
  display: flex;
  flex-direction: column;
  position: absolute;
  top: ${({ top, centerCapture  }) => centerCapture ? '50%' : `${top}px`};
  left: ${({ left, centreCapture }) => centreCapture ? '50%' : `${left}px`};
  width ${({ width }) => `${width}px`};
  background-color: ${({ backgroundColor }) => backgroundColor};
`;

export const HeroSectionTitle = styled.h2<any>`
  font-size: ${({ size }) => `${size}px`};
`;

export const HeroSectionDescription = styled.h5<any>`
  font-size: ${({ size }) => `${size}px`};
`;

const HeroSection = (props: HeroSectionProps) => {
 
  return (
    <HeroSectionWrapper {...props}>
      {props.backgroundImage && <HeroSectionImage src={props.backgroundImage} />}
      {props.capture && <HeroSectionCapture {...props.capture}>
        <HeroSectionTitle size={get(props, 'capture.titleSize')}>{get(props, 'capture.title')}</HeroSectionTitle>
        <HeroSectionDescription size={get(props, 'capture.DescriptionSize')}>{get(props, 'capture.description')}</HeroSectionDescription>
      </HeroSectionCapture>}
    </HeroSectionWrapper>
  );
}

const defaultProps = {
  backgroundColor: 'transparent',
  backgroundImage: 'https://images.unsplash.com/photo-1475351177616-1e5e440dccef?ixlib=rb-1.2.1&auto=format&fit=crop&w=2550&q=80',
};


const captureDefaultProps = {
  width: 400,
  backgroundColor: 'white',
}

HeroSectionCapture.defaultProps = captureDefaultProps;

HeroSection.defaultProps = defaultProps;

export default styled(HeroSection) `{
  width: ${({ width }) => width};
  height: ${({ height }) => height};
}`;
