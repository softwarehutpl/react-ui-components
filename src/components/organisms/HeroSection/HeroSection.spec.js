import React from 'react';
import { create, act } from 'react-test-renderer';
import 'jest-styled-components';
import HeroSection, {
  HeroSectionWrapper,
  HeroSectionImage,
  HeroSectionCapture,
  HeroSectionTitle,
  HeroSectionDescription,
} from './HeroSection.tsx';


describe('HeroSection Component', () => {
  it('should render HeroSection correctly', () => {
    const tree = create(<HeroSection />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render HeroSectionWrapper correctly', () => {
    const tree = create(<HeroSectionWrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render HeroSectionImage correctly', () => {
    const tree = create(<HeroSectionImage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render HeroSectionCapture correctly', () => {
    const tree = create(<HeroSectionCapture />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render HeroSectionTitle correctly', () => {
    const tree = create(<HeroSectionTitle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render HeroSectionDescription correctly', () => {
    const tree = create(<HeroSectionDescription />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render HeroSectionWrapper with default styles', () => {
    const component = create(<HeroSectionWrapper />);
    const accordionItemTitle = component.toJSON();
    expect(accordionItemTitle).toHaveStyleRule('display', 'flex');
    expect(accordionItemTitle).toHaveStyleRule('flex-direction', 'column');
    expect(accordionItemTitle).toHaveStyleRule('justify-content', 'flex-start');
    expect(accordionItemTitle).toHaveStyleRule('align-items', 'center');
  });

  it('should render HeroSectionImage with default styles', () => {
    const component = create(<HeroSectionImage />);
    const accordionItemTitle = component.toJSON();
    expect(accordionItemTitle).toHaveStyleRule('width', '100%');
    expect(accordionItemTitle).toHaveStyleRule('height', '100%');
    expect(accordionItemTitle).toHaveStyleRule('object-fit', 'contain');
  });

  it('should render HeroSectionWrapper with provided styles', () => {
    const component = create(
      <HeroSectionWrapper height="300px" width="800px" horizontalPadding={200} />
    );
    const accordionItemTitle = component.toJSON();
    expect(accordionItemTitle).toHaveStyleRule('width', '800px');
    expect(accordionItemTitle).toHaveStyleRule('height', '300px');
    expect(accordionItemTitle).toHaveStyleRule('padding', '0 200px');
  });
});
