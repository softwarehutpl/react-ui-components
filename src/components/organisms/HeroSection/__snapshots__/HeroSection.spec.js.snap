// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroSection Component should render HeroSection correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 undefinedpx;
}

.c1 {
  width: 100%;
  height: 100%;
  object-fit: contain;
}

<div
  className="c0 "
>
  <img
    className="c1"
    src="https://images.unsplash.com/photo-1475351177616-1e5e440dccef?ixlib=rb-1.2.1&auto=format&fit=crop&w=2550&q=80"
  />
</div>
`;

exports[`HeroSection Component should render HeroSectionCapture correctly 1`] = `
.c0 {
  width: display:flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
/>
`;

exports[`HeroSection Component should render HeroSectionDescription correctly 1`] = `
<h5
  className=""
/>
`;

exports[`HeroSection Component should render HeroSectionImage correctly 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  object-fit: contain;
}

<img
  className="c0"
/>
`;

exports[`HeroSection Component should render HeroSectionTitle correctly 1`] = `
<h3
  className=""
/>
`;

exports[`HeroSection Component should render HeroSectionWrapper correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 undefinedpx;
}

<div
  className="c0"
/>
`;
